database/tbcd/database.go

```diff
@@ -28,10 +49,13 @@ type Database interface {
 	MetadataPut(ctx context.Context, key, value []byte) error
 
 	// Block header
+	BlockHeaderBest(ctx context.Context) (*BlockHeader, error) // return canonical
 	BlockHeaderByHash(ctx context.Context, hash []byte) (*BlockHeader, error)
-	BlockHeadersBest(ctx context.Context) ([]BlockHeader, error)
+	BlockHeaderInsert(ctx context.Context, height uint64, bh [80]byte) error // do not use
```

Comment:
I think you could use a "Deprecated" comment here to prevent use and trigger warnings when used (https://staticcheck.io/docs/checks#SA1019): https://go.dev/wiki/Deprecated
```suggestion
	BlockHeaderInsert(ctx context.Context, height uint64, bh [80]byte) error // Deprecated: Use BlockHeadersInsert instead.
```